cli/tools/test.rs

```diff
@@ -388,14 +404,36 @@ impl TestReporter for PrettyTestReporter {
     if !summary.failures.is_empty() {
       println!("\nfailures:\n");
       for (description, error) in &summary.failures {
-        println!("{}", description.name);
+        println!(
+          "{} {} {}",
+          colors::gray(
+            self.to_relative_path_or_remote_url(&description.origin)
+          ),
+          colors::gray(">"),
+          description.name
+        );
         println!("{}", error);
         println!();
       }
 
-      println!("failures:\n");
+      let mut grouped_by_origin: HashMap<String, Vec<String>> =
+        HashMap::default();
```

Comment:
This output will have unstable sorting. Maybe use a `BTreeMap`?

Reply:
Yeah, I'll just do `BTreeMap`